kernel/kernel-defaults/src/test/java/io/delta/kernel/defaults/benchmarks/ReadMetadataRunner.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.kernel.defaults.benchmarks;
+
+import io.delta.kernel.Scan;
+import io.delta.kernel.ScanBuilder;
+import io.delta.kernel.Snapshot;
+import io.delta.kernel.Table;
+import io.delta.kernel.data.FilteredColumnarBatch;
+import io.delta.kernel.defaults.engine.DefaultEngine;
+import io.delta.kernel.engine.Engine;
+import io.delta.kernel.utils.CloseableIterator;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import org.apache.hadoop.conf.Configuration;
+import org.openjdk.jmh.infra.Blackhole;
+
+public class ReadMetadataRunner implements WorkloadRunner {
```

Comment:
nit: classdocs. How does this runner interelate to ReadMetadataSpec and WorkloadRunner?

Reply:
updated. PTAL